### PR TITLE
Dist build script and bits for OSX and Ubuntu

### PR DESCRIPTION
**Pull request by leet4tari**
Script to build archive with checksums and archive chechsum for OSX and Ubuntu (other platforms too follow)

ToDo: Add RedHat/CentOS/Fedora and other Linux distro - Upload dist archive on build

### CODE REVIEW

File: scripts/build-dists-tarball.sh [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  exit 2
+fi
+
+mkdir $distDir/dist
+
+cargo build --release
+
+COPY_FILES=(
+  "target/release/tari_base_node"
+  "config.toml"
```
Comment (CjS77) [inline scripts/build-dists-tarball.sh]:
```suggestion
  "config/presets/rincewind-simple.toml"
```

File: scripts/build-dists-tarball.sh [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  "config/tari_config_sample.toml"
+  "log4rs.yml"
```
Comment (CjS77) [inline scripts/build-dists-tarball.sh]: This file doesn't exist

File: scripts/build-dists-tarball.sh [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  "common/logging/log4rs-sample.yml"
+  "applications/tari_base_node/README.md"
+  "install.sh"
```
Comment (CjS77) [inline scripts/build-dists-tarball.sh]: This file doesn't exist

File: scripts/build-dists-tarball.sh [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+)
```
Comment (CjS77) [inline scripts/build-dists-tarball.sh]: You're missing a few files:

* scripts/install_tor.sh
* applications/tari_base_node/install-osx.sh
* applications/tari_base_node/start_tor.sh

File: scripts/build-dists-tarball.sh (hunk 1 above)
Comment (leet4tari) [reply scripts/build-dists-tarball.sh]: Can include ```config/presets/rincewind-simple.toml```

File: scripts/build-dists-tarball.sh (hunk 2 above)
Comment (leet4tari) [reply scripts/build-dists-tarball.sh]: ```config.toml``` nor ```log4rs.yml``` exist in the repo, but had been in the original request.  
  
I don't think we need to include ```log4rs.yml``` has this is now included in ```tari_base_node``` and can be create on first-run. I think keeping this in one place to maintain will be easier in the long run.  
  
Might be an idea to do the same for ```config.toml```?

File: scripts/build-dists-tarball.sh (hunk 4 above)
Comment (leet4tari) [reply scripts/build-dists-tarball.sh]: I am thinking that having each platform have their own folder for easier include and pointing users at in the repo, would make maintenance better.  
  
All three scripts are very OSX focused, but seem a little spread all over the repo.

File: scripts/build-dists-tarball.sh (hunk 2 above)
Comment (CjS77) [reply scripts/build-dists-tarball.sh]: There's a difference between the sample files (which typically list **all** possible options) and a minimal working file (which is what rincewind.toml is, for example).

Therefore Including `config/tari_config_sample.toml` is actually more for reference; it won't be used. It's the minimal working config that gets copied in the install script.

Comment (CjS77) [reply scripts/build-dists-tarball.sh]: In this case `log4rs-sample.yml`just so happens to double as a minimal config file.

File: scripts/build-dists-tarball.sh (hunk 4 above)
Comment (CjS77) [reply scripts/build-dists-tarball.sh]: Files in `scripts` are aimed at developers and are "meta" tools; i.e. they're used to help with the application / source code. So something like this script (`build-dists-tarball.sh`) belongs there.

Scripts that are application specific (`install_tor`, the base node install script) are kept with the specific application's source code.

The config files are platform-agnostic and are used by multiple applications, hence they're in the common crate.

By that measure, `install_tor` is really in the wrong place. The others are fine.

You're spot-on about the OSX-specific nature of some of the scripts; let's build towards multi-platform support; it doesn't all have to go in this PR.

Comment (CjS77) [reply scripts/build-dists-tarball.sh]: Can you elaborate on which files would be in separate folders?
Like 
* win/install.bat
* osx/install_sh

that sort of thing?

Comment (leet4tari) [reply scripts/build-dists-tarball.sh]: I would move
 - applications/tari_base_node/install-osx.sh -> ``osx/install.sh```
 - applications/tari_base_node/start_tor.sh -> ```osx/start_tor.sh```
 
and any other files or scripts that are ```osx``` or ```ubuntu``` or other platform specific.  
 
Then the dist build script can just include all the files in the platform folder into the archive.  
  
Don't think we will be able to do Win32/64 just yet, but I have not looked at cross-compiling in a long time.

Comment (CjS77) [reply scripts/build-dists-tarball.sh]: Yup, we can do that

**Pull request merged**